fix: fix tokens provider implementation - FE-4443 (#4540)
Sage/carbon: src/components/carbon-provider/carbon-provider.stories.mdx

A Carbon Provider uses the `ThemeProvider` from the [styled-components library](https://styled-components.com/docs/advanced#theming) to supply theme styles to your components. Themes are defined within the Carbon library.

# Design Tokens

> Carbon Provider provides also design tokens in form of CSS variables. It includes utilities that allow for mapping ThemeProvider themes to design token names and css variables.

Reviewer: ```suggestion
Carbon Provider also provides design tokens in the form of CSS variables. It includes utilities that map ThemeProvider themes to design token css variables.
```
